### PR TITLE
feature/sekoia_add_trigger_sekoiaio_case_comment_created

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/platform/issues/91875
Relates to https://github.com/SekoiaLab/integration/issues/1828

### Added

- Add a new `Case Comment Created` trigger for Operation Center cases, for enabling playbooks to react automatically when a comment is posted on a case (for example, forwarding analyst notes to third-party ticketing/case systems)

### Changed

- Improve case trigger test maintainability by refactoring repeated scenarios with `pytest.mark.parametrize`
- Significantly reduce case trigger test execution time while preserving behavior and assertions
- Refresh dependency lockfile (`poetry.lock`) with upgraded compatible package versions

### Fixed

- Increase reliability of case trigger behavior with additional tests covering edge cases (invalid sub-events, missing identifiers, API failures, and filter rejections)
- Reach 100% test coverage on `sekoiaio/triggers/cases.py`
- Improve case trigger resilience and troubleshooting with explicit API timeouts and richer error logs (request context included)

## Summary by Sourcery

Enable playbooks to react to newly created comments on Operation Center cases.

New Features:
- Add and register a Case Comment Created trigger that emits case and comment details for eligible Operation Center case comments.

Bug Fixes:
- Improve case trigger resilience with request timeouts, contextual API error logging, and expanded edge-case coverage.

Enhancements:
- Refactor and optimize case trigger tests while achieving complete coverage for case trigger handling.

Build:
- Refresh the Poetry dependency lockfile with compatible package updates.

Tests:
- Add comprehensive tests for comment-trigger handling, filtering, invalid events, missing identifiers, API failures, and response parsing errors.

### CODE REVIEW

File: Sekoia.io/sekoiaio/triggers/cases.py [hunk 1]
```diff
@@ -92,6 +93,48 @@ def _retrieve_case_from_caseapi(self, case_uuid):
...
+            try:
+                content = response.json()
+            except Exception:
+                content = response.text
+            self.log(
+                "Error while fetching case comment from Case API",
+                level="error",
+                status_code=response.status_code,
+                content=content,
+            )
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Sekoia.io/sekoiaio/triggers/cases.py:137]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/automation-library/pull/3023/changes/a086a06f11e29eea97da8d3ff9824d75a551c19f)

File: Sekoia.io/sekoiaio/triggers/cases.py [hunk 2]
```diff
@@ -92,6 +93,48 @@ def _retrieve_case_from_caseapi(self, case_uuid):
...
+        )
+        api_url = api_url.replace("/api/api", "/api")  # In case base_url ends with /api
+
+        api_key = self.module.configuration["api_key"]
+        headers = {"Authorization": f"Bearer {api_key}", "User-Agent": user_agent()}
+
+        response = requests.get(
+            api_url,
+            headers=headers,
+        )
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Sekoia.io/sekoiaio/triggers/cases.py:122]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/automation-library/pull/3023/changes/a086a06f11e29eea97da8d3ff9824d75a551c19f)

File: Sekoia.io/sekoiaio/triggers/cases.py [hunk 3]
```diff
@@ -255,3 +298,65 @@ def handle_event(self, message):
...
+        event = {
+            "comment": {
+                "uuid": comment.get("uuid"),
+                "content": comment.get("content"),
+                "author": comment.get("created_by"),
+                "date": comment.get("created_at"),
+            },
+            "case_uuid": case_uuid,
+            "short_id": case_short_id,
+            "status_uuid": case.get("status_uuid"),
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Sekoia.io/sekoiaio/triggers/cases.py]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/automation-library/pull/3023/changes/a086a06f11e29eea97da8d3ff9824d75a551c19f)

File: Sekoia.io/trigger_sekoiaio_case_comment_created.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+          "date": {
+            "type": "string",
+            "format": "date-time"
+          }
+        }
+      },
+      "case_uuid": {
+        "description": "Unique identifier of the Case (UUID string).",
+        "type": "string"
+      },
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Sekoia.io/trigger_sekoiaio_case_comment_created.json]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/automation-library/pull/3023/changes/a086a06f11e29eea97da8d3ff9824d75a551c19f)

**Review by mchupeau-sk — APPROVED** [review]
Nice ! LGFM
